packages/two_dimensional_scrollables/lib/src/tree_view/tree.dart

```diff
@@ -0,0 +1,1056 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'render_tree.dart';
+import 'tree_core.dart';
+import 'tree_delegate.dart';
+import 'tree_span.dart';
+
+// SHARED WITH FRAMEWORK - TreeViewNode (SliverTreeNode), TreeViewController (SliverTreeController)
+// Should not deviate from the core components of the framework.
+//
+// These classes share the same surface as SliverTree in the framework since
+// they are not currently available on the stable branch. After rolling to
+// stable, these classes may be deprecated, or more likely made to be
+// typedefs/subclasses of the framework core tree components. They could also
+// live on if at a later date the 2D TreeView deviates or adds special features
+// not relevant to the 1D sliver components of the framework.
+
+const double _kDefaultRowExtent = 40.0;
+
+/// A data structure for configuring children of a [TreeView].
+///
+/// A [TreeViewNode.content] can be of any type, but must correspond with the
+/// same type of the [TreeView].
+///
+/// Getters for [depth], [parent] and [isExpanded] are managed by the
+/// [TreeView]'s state.
+class TreeViewNode<T> {
+  /// Creates a [TreeViewNode] instance for use in a [TreeView].
+  TreeViewNode(
+    T content, {
+    List<TreeViewNode<T>>? children,
+    bool expanded = false,
+  })  : _expanded = children != null && children.isNotEmpty && expanded,
+        _content = content,
+        _children = children ?? <TreeViewNode<T>>[];
+
+  /// The subject matter of the node.
+  ///
+  /// Must correspond with the type of [TreeView].
+  T get content => _content;
+  final T _content;
+
+  /// Other [TreeViewNode]s this this node will be [parent] to.
```

Comment:
This might be something we could do... maybe like a childBuilder? We would still need to traverse the all active parts of the tree to get to a flat list of active nodes just to build it, since when we unpack the nodes we don't know if they are visible yet.

Reply:
I will file an issue for this feature, and we can see if this is something folks would like to have. 👍